src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -1,171 +1,560 @@
-                })
-                .style("opacity", 0.5)
-
-            scatter
-                .append("g")
-                .attr("class", "brush")
-                .call(brush);
-
-            var idleTimeout
-
-            function idled() {
-                idleTimeout = null;
+            brushElem.call(brush.move,null);
+        }
+
+        const zoomInKeyEvent = (event) => {
+            if ((event.ctrlKey || event.metaKey) && event.key === 'z') {
+                zoomIn();
+                selectedZoomBtn.style.display = "none";
             }
+        };
 
-            function updateChart() {
-                let extent = d3.event.selection;
-
-                if (!extent) {
-                    if (!idleTimeout) return idleTimeout = setTimeout(idled, 350);
-                    x.domain([4, 8])
-                } else {
-                    x.domain([x.invert(extent[0]), x.invert(extent[1])])
-                    scatter.select(".brush").call(brush.move, null)
-                }
-
-                xAxis.transition().duration(1000).call(d3.axisBottom(x))
-                scatter
-                    .selectAll("circle")
-                    .transition().duration(1000)
-                    .attr("cx", function (d) {
-                        return x(d[headerNames[0]]);
-                    })
-                    .attr("cy", function (d) {
-                        return y(d[headerNames[1]]);
-                    })
+        /**
+         * Updates plot when brushing.
+         */
+        function updateChart() {
+            let selectionEvent = d3.event.selection;
+            selectedZoomBtn.style.display = "inline-block";
+            selectedZoomBtn.addEventListener("click",zoomIn,true)
+            document.addEventListener('keydown',zoomInKeyEvent,true);
+            extent = selectionEvent;
+        }
 
-            }
+        /**
+         * Removes brush, keyboard event and zoom button when end event is captured.
+         */
+        const endBrushing = function (_) {
+            brushElem.call(brush.move,null);
+            selectedZoomBtn.style.display = "none";
+            selectedZoomBtn.removeEventListener("click",zoomIn,true)
+            document.removeEventListener('keydown',zoomInKeyEvent,true);
         };
+
+        let endEvents = ['click','auxclick','contextmenu','scroll']
+        endEvents.forEach(e => document.addEventListener(e,endBrushing,false));
     }
 
-    createDivElem(width, height) {
-        const divElem = document.createElementNS(null, "div");
-        divElem.setAttributeNS(null, "class", "vis-scatterplot");
-        divElem.setAttributeNS(null, "viewBox", 0 + " " + 0 + " " + width + " " + height);
-        divElem.setAttributeNS(null, "width", "100%");
-        divElem.setAttributeNS(null, "height", "100%");
-        divElem.setAttributeNS(null, "transform", "matrix(1 0 0 -1 0 0)");
+    /**
+     * Helper function for zooming into given scale.
```

Comment:
this doc doesnt explain what this function really does basedo n the input parameters. Please fix this and other docs - every doc should be very clear of what it does and why it needs the given inputs. Like here - this is a zooming into "given" scale. But the function does not consume a "scale", so how its given? Instead it conumes tons of strange parameters whose purpose is completely not mentioned in the docs. Please fix the docs in such way, that every function is clear about how it uses input parameters to compute something.